### DISCUSSION TITLE
Spark NLP 4.4.4: Patch release

### DISCUSSION THREAD

**Discussion by maziyarpanahi** [JohnSnowLabs/spark-nlp#13853] [Announcement]
---------------
:loudspeaker: Overview
---------------

Spark NLP 4.4.4 🚀 is a patch release with bug fixes and other improvements. We want to thank our community for their valuable feedback, feature requests, and contributions. Our [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) now contains over 17,000+ free and truly open-source models & pipelines. 🎉

### **Spark NLP** has a new home! https://sparknlp.org is where you can find all the documentation, models, and demos for Spark NLP. It aims to provide valuable resources to anyone interested in 100% open-source NLP solutions by using **Spark NLP** 🚀.

----------------
:star: New Features & Enhancements
----------------
* Add `Warmup` stage to loading all Transformers for word embeddings: ALBERT, BERT, CamemBERT, DistilBERT, RoBERTa, XLM-RoBERTa, and XLNet. This helps to reduce the first inference time and also validate importing external models from HuggingFace https://github.com/JohnSnowLabs/spark-nlp/pull/13851
* Add new notebooks to import ZeroShot Classifiers for Bert, DistilBERT, and RoBERTa fine-tuned based on NLI datasets https://github.com/JohnSnowLabs/spark-nlp/pull/13845

----------------
🐛 Bug Fixes
----------------
* Fix not being able to save models from XXXForSequenceClassitication and XXXForZeroShotClassification annotators https://github.com/JohnSnowLabs/spark-nlp/pull/13842
* Fix pretrained pipelines that stopped working since the 4.4.2 release on PySpark 3.2 and 3.3 versions (adding 121 new pipelines were added) https://github.com/JohnSnowLabs/spark-nlp/pull/13836

### :notebook: New Notebooks

| Notebooks | Colab | Colab |
| :-----------| :------|:------|
| BertForZeroShotClassification       | [HuggingFace in Spark NLP - BertForZeroShotClassification](https://github.com/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20BertForZeroShotClassification.ipynb)             | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20BertForZeroShotClassification.ipynb)       |
| DistilBertForZeroShotClassification       | [HuggingFace in Spark NLP - DistilBertForZeroShotClassification](https://github.com/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20DistilBertForZeroClassification.ipynb)             | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20DistilBertForZeroClassification.ipynb)       |
| RoBertaForZeroShotClassification       | [HuggingFace in Spark NLP - RoBertaForZeroShotClassification](https://github.com/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20RoBertaForZeroShotClassification.ipynb)             | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp/blob/master/examples/python/transformers/HuggingFace%20in%20Spark%20NLP%20-%20RoBertaForZeroShotClassification.ipynb)       |

- You can visit [Import Transformers in Spark NLP](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669)
- You can visit [Spark NLP Examples](https://github.com/JohnSnowLabs/spark-nlp/tree/master/examples) for 100+ examples

----------------
:book: Documentation
----------------
* [Import models from TF Hub & HuggingFace](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669) 
* [Spark NLP Notebooks](https://github.com/JohnSnowLabs/spark-nlp/tree/master/examples)
* [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) with new models
* [Spark NLP Articles](https://medium.com/spark-nlp)
* [Spark NLP in Action](https://nlp.johnsnowlabs.com/demo)
* [Spark NLP Documentation](https://nlp.johnsnowlabs.com/docs/en/quickstart)
* [Spark NLP Scala APIs](https://nlp.johnsnowlabs.com/api)
* [Spark NLP Python APIs](https://nlp.johnsnowlabs.com/api/python)

----------------
❤️  Community support
----------------

- [Slack](https://join.slack.com/t/spark-nlp/shared_invite/zt-198dipu77-L3UWNe_AJ8xqDk0ivmih5Q) For live discussion with the Spark NLP community and the team
- [GitHub](https://github.com/JohnSnowLabs/spark-nlp) Bug reports, feature requests, and contributions
- [Discussions](https://github.com/JohnSnowLabs/spark-nlp/discussions) Engage with other community members, share ideas,
  and show off how you use Spark NLP!
- [Medium](https://medium.com/spark-nlp) Spark NLP articles
- [YouTube](https://www.youtube.com/channel/UCmFOjlpYEhxf_wJUDuz6xxQ/videos) Spark NLP video tutorials

---------------
Installation
---------------
**Python**

```shell
#PyPI

pip install spark-nlp==4.4.4
```

**Spark Packages**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, and 3.3.x (Scala 2.12):

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp_2.12:4.4.4

pyspark --packages com.johnsnowlabs.nlp:spark-nlp_2.12:4.4.4
```

**GPU**

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:4.4.4

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:4.4.4
```

**Apple Silicon (M1 & M2)**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:4.4.4

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:4.4.4
```

**AArch64**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:4.4.4

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:4.4.4
```

**Maven**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, and 3.3.x:

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp_2.12</artifactId>
    <version>4.4.4</version>
</dependency>
```

**spark-nlp-gpu:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-gpu_2.12</artifactId>
    <version>4.4.4</version>
</dependency>
```

**spark-nlp-silicon:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-silicon_2.12</artifactId>
    <version>4.4.4</version>
</dependency>
```

**spark-nlp-aarch64:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-aarch64_2.12</artifactId>
    <version>4.4.4</version>
</dependency>
```

**FAT JARs**

* CPU on Apache Spark 3.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-assembly-4.4.4.jar

* GPU on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-gpu-assembly-4.4.4.jar

* M1 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-silicon-assembly-4.4.4.jar

* AArch64 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-aarch64-assembly-4.4.4.jar

## What's Changed
* Models hub by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13837
* FEATURE NMH-175: Add Copy to s3 on open source models [skip-test] by @KshitizGIT in https://github.com/JohnSnowLabs/spark-nlp/pull/13844
* FEATURE NMH-175: Remove models with missing s3 [skip-test] by @KshitizGIT in https://github.com/JohnSnowLabs/spark-nlp/pull/13847
* Resolve saving bug with multilabel parameter by @DevinTDHa in https://github.com/JohnSnowLabs/spark-nlp/pull/13842
* SPARKNLP-815: Add examples for ZeroShotClassifiers by @DevinTDHa in https://github.com/JohnSnowLabs/spark-nlp/pull/13845
* SPARKNLP 801 set up warmup for all embeddings by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13851
* Sparknlp 801 set up warmup for all embeddings classifiers by @maziyarpanahi in https://github.com/JohnSnowLabs/spark-nlp/pull/13852

**Full Changelog**: https://github.com/JohnSnowLabs/spark-nlp/compare/4.4.3...4.4.4

<hr /><em>This discussion was created from the release <a href='https://github.com/JohnSnowLabs/spark-nlp/releases/tag/4.4.4'>Spark NLP 4.4.4: Patch release</a>.</em>
